# SR_XAI Functional Toy Test

This notebook records a dependency-light functional test for the local explanation optimizer idea. The test creates a synthetic 16x16x16 Gaussian target, generates nine noisy explanation maps, computes simple faithfulness and complexity scores, and fuses the maps with metric-derived weights.

In [1]:
import json
from toy_xai_pipeline import run_toy_pipeline

results = run_toy_pipeline("results")
print(json.dumps(results["fused_metrics"], indent=2))

{
  "faithfulness": 0.9629136755999348,
  "complexity": 0.04443359375
}


## Summary

- Dataset: `synthetic_16x16x16_gaussian`
- Explainers: 9
- Fused faithfulness: `0.9629136755999348`
- Fused complexity: `0.04443359375`

The fused explanation is more faithful than the best single synthetic explainer in this toy setup, while remaining sparse.

In [2]:
[(row["name"], row["faithfulness"], row["complexity"]) for row in results["individual_metrics"]]

[('DeepLiftshap', 0.9250594673395487, 0.053955078125),
 ('KernelShape', 0.8693542150128789, 0.117431640625),
 ('Lime', 0.8201613606973509, 0.33203125),
 ('GradShap', 0.7623334857653209, 0.275390625),
 ('Saliency', 0.7153108354391768, 0.572021484375),
 ('Intgrag', 0.6532185372584012, 0.428955078125),
 ('DeepLift', 0.6045666510638368, 0.777099609375),
 ('GuidedBackprop', 0.5701082996794089, 0.838623046875),
 ('GuidedGradCam', 0.536169980538578, 0.80029296875)]